tests/visual/test_show_map.py

```diff
@@ -126,3 +126,15 @@ def test_show_map_crs_change_to_4326_2(layer_polygon, layer_points, layer_polygo
     QgsProject.instance().addMapLayers(
         [layer_points, layer_polygon_3067, layer_polygon]
     )
+
+
+@pytest.mark.qgis_show_map(timeout=DEFAULT_TIMEOUT, zoom_to_common_extent=False)
+def test_show_map_should_not_keep_the_set_extent(
+    layer_polygon_3067, qgis_canvas, qgis_app
+):
+    QgsProject.instance().addMapLayer(layer_polygon_3067)
+    qgis_canvas.setExtent(QgsRectangle(475804, 7145949.5, 549226, 7219371.5))
+    # This triggers the map to set the extent based on the layer
+    # if events are not processed after adding the layer
+    qgis_app.processEvents()
+    assert qgis_canvas.extent().area() < 1e11  # noqa: PLR2004
```

Comment:
Should the extent here be based on the one set explicitly on line 136?
Could you please comment what is this 1e11? Or maybe extract it to a describing variable.

Reply:
I replaced this to compare the heights of the rectangle and the extent. The width might differ a bit so the rectangles cannot be compared directly. I am not sure what causes the widths to differ but I think it is not in scope of this PR.